core/services/pg/lease_lock.go

```diff
@@ -81,7 +80,7 @@ func NewLeaseLock(db *sqlx.DB, appID uuid.UUID, lggr logger.Logger, refreshInter
 
 // TakeAndHold will block and wait indefinitely until it can get its first lock
 // NOT THREAD SAFE
-func (l *leaseLock) TakeAndHold(sig shutdown.Signal) (err error) {
+func (l *leaseLock) TakeAndHold() (err error) {
```

Comment:
Perhaps we want to pass the context in here? Since otherwise, we will not gracefully exit on sigterm if we are in the initial loop trying to take the lock

Reply:
Same as for advisory lock, the lock's contract is to call `Release()`. Otherwise we inject another side effect in here.
My initial implementation was using context instead of stop channel + waitgroup, but I wanted to control when I call `Release`. If we inject context here, we shall be removing `Release`, right? Is this what you want me to do?